### PR TITLE
[FIX] ScatterPlot: Fix crash when coloring by column of unknowns

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
ValueError cannot convert float NaN to integer owscatterplot
In case when all column values are NaN then it throws that error.
https://sentry.io/biolab/orange3/issues/196744604/

##### Description of changes
Logarithm of zero and division by zero are prevented.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/196744604/

File: Orange/widgets/visualize/tests/test_owscatterplot.py [hunk 1]
```diff
@@ -102,3 +103,26 @@ def test_report_on_empty(self):
...
         self.widget.report_caption.assert_not_called()
         self.widget.report_items.assert_not_called()
+
+    def test_data_column_nans(self):
+        """
+        ValueError cannot convert float NaN to integer.
+        In case when all column values are NaN then it throws that error.
+        GH-2061
+        """
+        dat = """\
```
Comment (astaric) [inline Orange/widgets/visualize/tests/test_owscatterplot.py]: Is there a special reason why the dataset is generated in this way? (and not with table like in https://github.com/biolab/orange3/pull/2057/files#diff-1a66c8eae521f07f1c5d904e0cd025a7R117)?

File: Orange/widgets/visualize/tests/test_owscatterplot.py [hunk 2]
```diff
@@ -102,3 +103,26 @@ def test_report_on_empty(self):
...
+            42.11\t\tneumann
+            17.01\t\tneumann
+            19.01\t\tneumann
+            33.21\t\tturing
+            34.01\t\tturing"""
+        file = io.StringIO(dat)
+        data = TabReader(file).read()
+        self.send_signal("Data", data)
+
+        self.widget.graph.attr_color = self.widget.get_shown_attributes()[1]
```
Comment (astaric) [inline Orange/widgets/visualize/tests/test_owscatterplot.py]: Could you select attributes using helpers found in https://github.com/biolab/orange3/blob/master/Orange/widgets/tests/utils.py#L268?

File: Orange/widgets/visualize/tests/test_owscatterplot.py (hunk 1 above)
Comment (jerneju) [inline Orange/widgets/visualize/tests/test_owscatterplot.py]: No particular reason at all. Now changed.

File: Orange/widgets/visualize/owscatterplot.py [hunk 3]
```diff
@@ -409,7 +409,7 @@ def handleNewSignals(self):
         self.graph.new_data(self.data_metas_X, self.subset_data)
         if self.attribute_selection_list and \
                 all(attr in self.graph.domain
-                    for attr in self.attribute_selection_list):
+                        for attr in self.attribute_selection_list):
```
Comment (kernc) [inline Orange/widgets/visualize/owscatterplot.py]: This was indented ok before.

Comment (jerneju) [inline Orange/widgets/visualize/owscatterplot.py]: Pylint complained and that is why I added 4 spaces.

**Pull request merged**